backend/app/api/endpoints/adapter/chat.py

```
@@ -490,3 +490,178 @@ async def apply_correction(
+class FixMermaidResponse(BaseModel):
+    """Response body for Mermaid diagram fix."""
+
+    success: bool
+    fixed_code: Optional[str] = None
+    error: Optional[str] = None
+
+
+@router.post("/fix-mermaid")
+async def fix_mermaid_code(
+    request: FixMermaidRequest,
+    db: Session = Depends(get_db),
```

Ruff (0.14.13)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)